Commands/Files/AddFile.cs

```diff
@@ -124,6 +132,47 @@ protected override void ExecuteCmdlet()
 
                 ClientContext.ExecuteQueryRetry();
             }
+            if (ContentType != null)
+            {
+                ClientContext.Load(file, f=>f.ListId);
+                ClientContext.ExecuteQueryRetry();
+
+                List list = SelectedWeb.Lists.GetById(file.ListId);
+                ClientContext.Load(list);
+                ClientContext.ExecuteQueryRetry();
+
+                ContentType ct;
+                if (!string.IsNullOrEmpty(ContentType.Id))
+                {
+                    ct = list.GetContentTypeById(ContentType.Id);
+                }
+                else if(!string.IsNullOrEmpty(ContentType.Name))
+                {
+                    ct = list.GetContentTypeByName(ContentType.Name);
+                }
+                else if(ContentType.ContentType != null)
+                {
+                    ct = ContentType.ContentType;
+                }
+                else
+                {
+                    ct = null;
+                   
+                }
+
+                if (ct != null)
+                {
+                    var item = file.ListItemAllFields;
+                    item["ContentTypeId"] = ct.Id.StringValue;
+                    item.Update();
+                    ClientContext.ExecuteQueryRetry();
+                }
+                else
+                {
+                    WriteVerbose("Content Type was 'null' was not found.");
```

Comment:
Thanks for the links. This seems to be working, but as I am testing I  noticed a strange error. 

When I run both commands  

Get-PnPContentType -Identity Documents 
Add-PnPFile -Folder /TestLib/TestFolder -Path ./Testfile.txt -ContentType Documents

I get a [format-default] collection not initialized.. but when I run them independently it all works no problem..